nixos/modules/services/misc/n8n.nix

```diff
@@ -0,0 +1,433 @@
+          };
+
+          hash = mkOption {
+            type = types.bool;
+            default = false;
+            description = "If password for basic auth is hashed";
+          };
+        };
+
+        jwtAuth = {
+          active = mkEnableOption "If jwt auth should be activated for editor and REST-API";
+
+          jwtHeader = mkOption {
+            type = types.str;
+            default = "";
+            description = "The request header containing a signed JWT";
+          };
+
+          jwtHeaderValuePrefix = mkOption {
+            type = types.str;
+            default = "";
+            description = "The request header value prefix to strip (optional)";
+          };
+
+          jwksUri = mkOption {
+            type = types.str;
+            default = "";
+            description = "The URI to fetch JWK Set for JWT authentication";
+          };
+
+          jwtIssuer = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT issuer to expect (optional)";
+          };
+
+          jwtNamespace = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT namespace to expect (optional)";
+          };
+
+          jwtAllowedTenantKey = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT tenant key name to inspect within JWT namespace (optional)";
+          };
+
+          jwtAllowedTenant = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT tenant to allow (optional)";
+          };
+        };
+      };
+
+      endpoints = {
+        rest = mkOption {
+          type = types.str;
+          default = "rest";
+          description = "Path for rest endpoint";
+        };
+
+        webhook = mkOption {
+          type = types.str;
+          default = "webhook";
+          description = "Path for webhook endpoint";
+        };
+
+        webhookTest = mkOption {
+          type = types.str;
+          default = "webhook-test";
+          description = "Path for test-webhook endpoint";
+        };
+      };
+
+      externalHookFiles = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = "Files containing external hooks. Multiple files can be separated by colon (\":\")";
+      };
+
+      nodes = {
+        exclude = mkOption {
+          type = types.listOf types.str;
+          default = [];
+          description = "Nodes not to load";
+        };
+
+        errorTriggerType = mkOption {
+          type = types.str;
+          default = "n8n-nodes-base.errorTrigger";
+          description = "Node Type to use as Error Trigger";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.n8n = {
+      description = "N8N service";
+      after = [ "target.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
```

Comment:
Bonus points awarded if you can apply some basic `systemd` hardening.

Reply:
Any pointer to suggest?

Reply:
[This](https://gist.github.com/ageis/f5595e59b1cddb1513d1b425a323db04) is a nice read, if you're interested.

Reply:
Thank you

Reply:
**Full** bonus points awarded :sparkles: